.github/workflows/comment.yaml

```diff
@@ -1,42 +1,99 @@
 name: Comment on PR
 
 on:
-  workflow_call:
-    inputs:
-      libraryName:
-        type: string
-        required: false
-      resultsFile:
-        required: true
-        type: string
-    secrets:
-      token:
-        required: true
+  workflow_run:
+    workflows:
+      - "Test"
+      - "Absinthe Federation Test"
+      - "Apollo Server Test"
+      - "Ariadne Test"
+      - "AWS AppSync Test"
+      - "Caliban Test"
+      - "Dgraph Test"
+      - "DGS Test"
+      - "Express GraphQL Test"
+      - "Federation JVM Test"
+      - "GQLGen Test"
+      - "Graphene Test"
+      - "GraphQL for .Net Test"
+      - "GraphQL Go Test"
+      - "GraphQL Helix Test"
+      - "GraphQL Java Kickstart Test"
+      - "GraphQL Kotlin Test"
+      - "GraphQL Mesh Test"
+      - "GraphQL Yoga Test"
+      - "HotChocolate Test"
+      - "Lighthouse Test"
+      - "Mercurius Test"
+      - "NestJS (Code First) Test"
+      - "NestJS (SDL First) Test"
+      - "Apollo Federation PHP Test"
+      - "Pothos Test"
+      - "GraphQL Ruby Test"
+      - "Sangria Test"
+      - "StepZen Test"
+      - "Strawberry GraphQL Test"
+    types:
+      - completed
 
 jobs:
   comment:
     timeout-minutes: 30
     runs-on: ubuntu-latest
+    if: >
+      github.event.workflow_run.event == 'pull_request' &&
+      github.event.workflow_run.conclusion == 'success'
     steps:
-      - name: Download Results File
-        uses: actions/download-artifact@v3
+      - name: Download Artifacts
+        uses: actions/github-script@v6
+        id: download_artifacts
         with:
-          name: ${{ inputs.resultsFile }}
+          script: |
```

Comment:
not a big fan of this script but sadly there is no easy way to share those artifacts... sadly we cannot use [dispatch events](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#repository_dispatch) as they require token with WRITE permissions (so again same problem as this PR is attempting to solve).

script is adapted from https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

Reply:
we also cannot just use `actions/download-artifact` to download all artifacts as it does not support downloading artifacts from other workflows...